### PR TITLE
Implementation of Robotoff insight annotation interface

### LINKED ISSUE #2234: Ask questions about incomplete products
Ask simple questions to users about incomplete products

https://github.com/shivasurya/FeedbackDialog

For barcode: 00024907 and if the UX is in French
https://robotoff.openfoodfacts.org/api/v1/questions/00024907?lang=fr

Then depending on the answer, POST request:
https://robotoff.openfoodfacts.org/api/v1/insights/annotate

The request type must be application/x-www-form-urlencoded.
```
- insight_id (str, required) - ID of the insight
- annotation (int, required) - Annotation of the prediction: 1 to accept the prediction, 0 to refuse it, and -1 for "unknown".
- update (int, optional) - Send the update to Openfoodfacts if update=1, don't send the update otherwise. This parameter is useful if the update is performed client-side.
```

### PR DESCRIPTION
**Pull request by raphael0202**
## Description

Similar to PR https://github.com/openfoodfacts/openfoodfacts-androidapp/pull/2302, with the following differences:
- Use rxjava to deal with asynchronous calls (much cleaner code)
- the question is not displayed at the bottom of the screen but is inlined in the product view
- if the question is about the product category, the "Add a category to compute the nutriscore" button is hidden

## Related issues and discussion
#fixes #2234
 
 ## Screen-shots, if any
 
photo5828025391380541365
photo5828025391380541364
photo5828025391380541363

 ## Checklist
 
Make sure you've done all the following (You can delete the checklist before submitting)
 
 - [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 - [ ] Code is well documented
 - [ ] Include unit tests for new functionality
 - [ ] All user-visible strings are made translatable
 - [ ] Code passes Travis builds in your branch
 - [ ] If you have multiple commits please combine them into one commit by squashing them.
 - [ ] Read and understood the contribution guidelines .

### CODE REVIEW

File: app/src/main/res/layout/fragment_summary_product.xml [hunk 1]
```diff
@@ -291,6 +291,73 @@
...
+                    >
+                    <TextView
+                        android:id="@+id/product_question_text"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:textColor="@android:color/white"
+                        android:paddingRight="@dimen/spacing_small"
+                        android:paddingLeft="@dimen/spacing_small"
+                        android:paddingTop="@dimen/spacing_small"
+                        tools:text="Does the product belong to this category?"
```
Comment (teolemon) [inline app/src/main/res/layout/fragment_summary_product.xml]: translatable ?

Comment (raphael0202) [reply app/src/main/res/layout/fragment_summary_product.xml]: `tools:text` attributes is for development display only

**Comment by teolemon** [comment]
Related to server side issues (the db is not updated in real time):
image

**Comment by teolemon** [comment]
@subhanjansaha @raphael0202 Congratulations to both on this PR :-)
I'm merging. This will have some impact I think :)

**Pull request merged**